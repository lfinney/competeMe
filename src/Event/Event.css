.Event {
  background-color: #7a7d7d;
  border-radius: 10px;
  margin: 0px 15px 15px 15px;
  box-shadow: 5px 5px 0px 1px rgba(26, 26, 26, 0.75); }

.Event h3, h4, p {
  margin: 10px 0; }

.Event .event-info {
  display: grid;
  grid-template-columns: auto 250px 125px;
  grid-template-areas: "info map button"; }

.Event .event-test-details {
  grid-area: info;
  display: flex;
  flex-direction: column;
  justify-content: flex-start; }

.Event .park-map {
  width: 200px;
  height: 200px;
  background-color: #1a1a1a;
  margin: 25px;
  grid-area: map; }

.Event button {
  width: 100px;
  height: 100px;
  margin-top: 75px;
  grid-area: button; }
