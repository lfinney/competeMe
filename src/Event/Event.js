import React from 'react';
const dateFormat = require('dateformat');
const convertTime = require('convert-time');



const Event = ({ comp }) => {
  console.log(comp);
  return (
    <div className="Event">
      <h3>{comp.compName}</h3>
      <h4>{comp.sport} - {comp.competitiveness}</h4>
      <h4>{comp.players}</h4>
      <h4>{dateFormat(comp.date, 'mediumDate')}, {convertTime(comp.time)}, {comp.location}</h4>
      <p>{comp.details}</p>
      <div className="park-map"></div>
      <button>Count Me In!</button>
    </div>
  );
};


export default Event;
