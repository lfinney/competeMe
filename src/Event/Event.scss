@import '../variables.scss';

.Event {
  background-color: $gray;
  border-radius: $border-radius;
  margin: 0px 15px 15px 15px;
  box-shadow: 5px 5px 0px 1px rgba(26, 26, 26, .75);
}

.Event h3, h4, p {
  margin: 10px 0;
}

.Event .event-info {
  display: grid;
  grid-template-columns: auto 250px 125px;
  grid-template-areas: "info map button"
}

.Event .event-test-details {
  grid-area: info;
  display: flex;
  flex-direction: column;
  justify-content: flex-start;
}

.Event .park-map {
  width: 200px;
  height: 200px;
  background-color: $black;
  margin: 25px;
  grid-area: map;
}

.Event button {
  width: 100px;
  height: 100px;
  margin-top: 75px;
  grid-area: button;
}
