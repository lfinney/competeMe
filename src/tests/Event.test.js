import React from 'react';
import ReactDOM from 'react-dom';
import Event from '../Event/Event';
import { shallow } from 'enzyme';


describe('Event', () => {

  it('should render component', () => {
    const wrapper = shallow(
      <Event />);

    expect(wrapper.length).toEqual(1);
  });
});
