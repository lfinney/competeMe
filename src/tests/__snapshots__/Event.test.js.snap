// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Event should render component 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Event
    comp={
        Object {
            "compName": "Frisbee",
            "competitiveness": "casual",
            "date": "1987-10-09",
            "details": "This will be fun",
            "location": "Washington Park",
            "player": 7,
            "sport": "Frisbee",
            "time": "15:00",
          }
    }
/>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": null,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="event-info"
>
        <div
                className="event-text-details"
        >
                <h4>
                        Frisbee
                         - 
                        casual
                </h4>
                <h4 />
                <h4>
                        3:00 pm 
                        on 
                        Oct 8, 1987 
                        at 
                        Washington Park
                </h4>
                <p>
                        This will be fun
                </p>
                <h3>
                        Frisbee
                </h3>
        </div>
        <div
                className="park-map"
        />
        <button
                onClick={[Function]}
        >
                Count Me In!
        </button>
</div>,
      "className": "Event",
      "liveUser": undefined,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="event-text-details"
>
            <h4>
                        Frisbee
                         - 
                        casual
            </h4>
            <h4 />
            <h4>
                        3:00 pm 
                        on 
                        Oct 8, 1987 
                        at 
                        Washington Park
            </h4>
            <p>
                        This will be fun
            </p>
            <h3>
                        Frisbee
            </h3>
</div>,
          <div
            className="park-map"
/>,
          <button
            onClick={[Function]}
>
            Count Me In!
</button>,
        ],
        "className": "event-info",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <h4>
                Frisbee
                 - 
                casual
</h4>,
              <h4 />,
              <h4>
                3:00 pm 
                on 
                Oct 8, 1987 
                at 
                Washington Park
</h4>,
              <p>
                This will be fun
</p>,
              <h3>
                Frisbee
</h3>,
            ],
            "className": "event-text-details",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "Frisbee",
                  " - ",
                  "casual",
                ],
              },
              "ref": null,
              "rendered": Array [
                "Frisbee",
                " - ",
                "casual",
              ],
              "type": "h4",
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "h4",
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": Array [
                  "3:00 pm ",
                  "on ",
                  "Oct 8, 1987 ",
                  "at ",
                  "Washington Park",
                ],
              },
              "ref": null,
              "rendered": Array [
                "3:00 pm ",
                "on ",
                "Oct 8, 1987 ",
                "at ",
                "Washington Park",
              ],
              "type": "h4",
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": "This will be fun",
              },
              "ref": null,
              "rendered": "This will be fun",
              "type": "p",
            },
            Object {
              "instance": null,
              "key": null,
              "nodeType": "host",
              "props": Object {
                "children": "Frisbee",
              },
              "ref": null,
              "rendered": "Frisbee",
              "type": "h3",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "className": "park-map",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": null,
          "nodeType": "host",
          "props": Object {
            "children": "Count Me In!",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": "Count Me In!",
          "type": "button",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": null,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="event-info"
>
          <div
                    className="event-text-details"
          >
                    <h4>
                              Frisbee
                               - 
                              casual
                    </h4>
                    <h4 />
                    <h4>
                              3:00 pm 
                              on 
                              Oct 8, 1987 
                              at 
                              Washington Park
                    </h4>
                    <p>
                              This will be fun
                    </p>
                    <h3>
                              Frisbee
                    </h3>
          </div>
          <div
                    className="park-map"
          />
          <button
                    onClick={[Function]}
          >
                    Count Me In!
          </button>
</div>,
        "className": "Event",
        "liveUser": undefined,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": null,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="event-text-details"
>
              <h4>
                            Frisbee
                             - 
                            casual
              </h4>
              <h4 />
              <h4>
                            3:00 pm 
                            on 
                            Oct 8, 1987 
                            at 
                            Washington Park
              </h4>
              <p>
                            This will be fun
              </p>
              <h3>
                            Frisbee
              </h3>
</div>,
            <div
              className="park-map"
/>,
            <button
              onClick={[Function]}
>
              Count Me In!
</button>,
          ],
          "className": "event-info",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <h4>
                  Frisbee
                   - 
                  casual
</h4>,
                <h4 />,
                <h4>
                  3:00 pm 
                  on 
                  Oct 8, 1987 
                  at 
                  Washington Park
</h4>,
                <p>
                  This will be fun
</p>,
                <h3>
                  Frisbee
</h3>,
              ],
              "className": "event-text-details",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "Frisbee",
                    " - ",
                    "casual",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "Frisbee",
                  " - ",
                  "casual",
                ],
                "type": "h4",
              },
              Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "children": undefined,
                },
                "ref": null,
                "rendered": null,
                "type": "h4",
              },
              Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "children": Array [
                    "3:00 pm ",
                    "on ",
                    "Oct 8, 1987 ",
                    "at ",
                    "Washington Park",
                  ],
                },
                "ref": null,
                "rendered": Array [
                  "3:00 pm ",
                  "on ",
                  "Oct 8, 1987 ",
                  "at ",
                  "Washington Park",
                ],
                "type": "h4",
              },
              Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "children": "This will be fun",
                },
                "ref": null,
                "rendered": "This will be fun",
                "type": "p",
              },
              Object {
                "instance": null,
                "key": null,
                "nodeType": "host",
                "props": Object {
                  "children": "Frisbee",
                },
                "ref": null,
                "rendered": "Frisbee",
                "type": "h3",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "className": "park-map",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": null,
            "nodeType": "host",
            "props": Object {
              "children": "Count Me In!",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Count Me In!",
            "type": "button",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
